NitroxServer/Communication/ConnectionState.cs

```diff
@@ -1,11 +1,32 @@
-﻿namespace NitroxServer.Communication
+﻿using LiteNetLib;
+
+namespace NitroxServer.Communication;
+
+public enum NitroxConnectionState
 {
-    public enum NitroxConnectionState
+    Unknown,
+    Disconnected,
+    Connected,
+    Reserved,
+    InGame
+}
+
+public static class NitroxConnectionStateExtension
+{
+    public static NitroxConnectionState ToNitrox(this ConnectionState connectionState)
     {
-        Unknown,
-        Disconnected,
-        Connected,
-        Reserved,
-        InGame
+        NitroxConnectionState state = NitroxConnectionState.Unknown;
+
+        if (connectionState.HasFlag(ConnectionState.Connected))
+        {
+            state = NitroxConnectionState.Connected;
+        }
+
+        if (connectionState.HasFlag(ConnectionState.Disconnected))
+        {
+            state = NitroxConnectionState.Disconnected;
+        }
+
+        return state;
     }
 }
```

Comment:
Would use bitwise operations here because HasFlag isn't generic and boxes the Enum argument (fixed in .NET Core+, but we're on mono)
Extension classes usually end with plural `s`

```suggestion
public static class NitroxConnectionStateExtensions
{
    public static NitroxConnectionState ToNitrox(this ConnectionState connectionState)
    {
        if ((connectionState & ConnectionState.Connected) == ConnectionState.Connected)
        {
            return NitroxConnectionState.Connected;
        }
        if ((connectionState & ConnectionState.Disconnected) == ConnectionState.Disconnected)
        {
            return NitroxConnectionState.Disconnected;
        }

        return NitroxConnectionState.Unknown;
    }
}
```